x-pack/plugins/security/server/audit/audit_service.ts

```diff
@@ -5,53 +5,176 @@
-      getLogger: (id?: string): AuditLogger => {
-        return {
-          log: (eventType: string, message: string, data?: Record<string, any>) => {
-            if (!this.auditLoggingEnabled) {
-              return;
-            }
-
-            this.logger.info(message, {
-              tags: id ? [id, eventType] : [eventType],
-              eventType,
-              ...data,
-            });
+    // Configure logging
+    logging.configure(license.features$.pipe(createLoggingConfig(config)));
+
+    /**
+     * Creates an {@link AuditLogger} scoped to the current request.
+     *
+     * @example
+     * ```typescript
+     * const auditLogger = securitySetup.audit.asScoped(request);
+     * auditLogger.log(event);
+     * ```
+     */
+    const asScoped = (request: KibanaRequest): AuditLogger => {
+      /**
+       * Logs an {@link AuditEvent} and automatically adds meta data about the
+       * current user, space and correlation id.
+       *
+       * Guidelines around what events should be logged and how they should be
+       * structured can be found in: `/x-pack/plugins/security/README.md`
+       *
+       * @example
+       * ```typescript
+       * const auditLogger = securitySetup.audit.asScoped(request);
+       * auditLogger.log({
+       *   message: 'User is updating dashboard [id=123]',
+       *   event: {
+       *     action: 'saved_object_update',
+       *     outcome: 'unknown'
+       *   },
+       *   kibana: {
+       *     saved_object: { type: 'dashboard', id: '123' }
+       *   },
+       * });
+       * ```
+       */
+      const log = (event: AuditEvent) => {
+        const user = getCurrentUser(request);
+        const spaceId = getSpaceId(request);
+        const meta: AuditLogMeta = {
+          ...event,
+          user:
+            (user && {
+              name: user.username,
+              roles: user.roles,
+            }) ||
+            event.user,
+          kibana: {
+            space_id: spaceId,
+            ...event.kibana,
+          },
+          trace: {
+            id: request.id,
           },
         };
-      },
+        if (filterEvent(meta, config.ignore_filters)) {
+          this.betaLogger.info(event.message!, meta);
+        }
+      };
+      return { log };
     };
+
+    /**
+     * @deprecated
+     * Use `audit.asScoped(request)` method instead to create an audit logger
+     */
+    const getLogger = (id?: string): LegacyAuditLogger => {
+      return {
+        log: (eventType: string, message: string, data?: Record<string, any>) => {
+          if (!this.allowAuditLogging) {
+            return;
+          }
+
+          this.logger.info(message, {
+            tags: id ? [id, eventType] : [eventType],
+            eventType,
+            ...data,
+          });
+        },
+      };
+    };
+
+    http.registerOnPostAuth((request, response, t) => {
+      if (request.auth.isAuthenticated) {
+        asScoped(request).log(httpRequestEvent({ request }));
```

Comment:
Part 2 of my suggested change above.
```suggestion
        const spaceId = getSpaceId(request);
        asScoped(request).log(httpRequestEvent({ request }, spaceId));
```